src/Cake.Common/Solution/SolutionParser.cs

```diff
@@ -182,5 +195,104 @@ private static void ParseNestedProjectLine(List<SolutionProject> projects, strin
             parent.Items.Add(child);
             child.Parent = parent;
         }
+
+        private static SolutionParserResult ParseSlnxSolution(IFile file)
+        {
+            var xmlDocument = new XmlDocument();
+            using var xmlContentStream = file.OpenRead();
+            xmlDocument.Load(xmlContentStream);
+
+            var root = xmlDocument.DocumentElement;
+            if (root == null)
+            {
+                throw new CakeException($"Solution file '{file.Path.FullPath}' does not contain a root element.");
+            }
+
+            var projects = new List<SolutionProject>();
+            foreach (var xmlElement in root.ChildNodes.OfType<XmlElement>())
+            {
+                var projectsFromElement = ParseSlnxElement(xmlElement, file);
+                projects.AddRange(projectsFromElement);
+            }
+
+            // the new SLNX has no information about the file format version, VS version and minimal VS version, so it's omitted.
+            return new SolutionParserResult(string.Empty, string.Empty, string.Empty, projects);
```

Comment:
ParseSlnSolution uses projects.AsReadOnly() (line 125), but ParseSlnxSolution passes a mutable list. This breaks immutability and can potentially cause runtime issues.
```suggestion
            return new SolutionParserResult(string.Empty, string.Empty, string.Empty, projects.AsReadOnly());
```

Reply:
Good point, thanks. This slipped through. Addressed this with my latest changes.